apollo-federation/src/sources/connect/json_selection/parser.rs

```diff
@@ -67,9 +67,17 @@ pub enum NamedSelection {
 impl NamedSelection {
     fn parse(input: &str) -> IResult<&str, Self> {
         alt((
+            // We must try parsing NamedPathSelection before NamedFieldSelection
+            // and NamedQuotedSelection because a NamedPathSelection without a
+            // leading `.`, such as `alias: some.nested.path` has a prefix that
+            // can be parsed as a NamedFieldSelection: `alias: some`. Parsing
+            // then fails when it finds the remaining `.nested.path` text. Some
+            // parsers would solve this by forbidding `.` in the "lookahead" for
+            // Named{Field,Quoted}Selection, but negative lookahead is tricky in
+            // nom, so instead we greedily parse NamedPathSelection first.
+            Self::parse_path,
             Self::parse_field,
             Self::parse_quoted,
-            Self::parse_path,
             Self::parse_group,
```

Comment:
This is the operative change that fixes the `NamedPathSelection` parsing ambiguity.